autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -50,31 +49,40 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         long id = event.getClusterId();
         LOGGER.info("Cluster {} failed", id);
         Cluster cluster = clusterService.find(id);
+        if (cluster == null) {
```

Comment:
its a weird case when the cluster null, maybe logging it would be useful

Reply:
There can be multiple evaluators in the threadpool. Let's suppose there is a cluster that is being terminated. All evaluators belonging to the cluster will timeout when connecting ambari => each of them will raise and UpdateFailedEvent.

Handling the first will check cluster status, will see that it is being deleted, so deletes the cluster from periscope. Irrespective of this, all the other evaluators will also raise the UpdateFailedEvent. Whenever they enter the handler, there will be no cluster in periscope any more.

Given this scenario, do you still suggest to insert log lines?

Reply:
No, I was only guessing it could be useful, but I don't know this part of the code as well as you, so I'll leave it to your judgement!